# Generated by Django 5.1.4 on 2024-12-29 10:53

import django.utils.timezone
from django.db import migrations, models


class Migration(migrations.Migration):

    initial = True

    dependencies = []

    operations = [
        migrations.CreateModel(
            name="Error",
            fields=[
                (
                    "id",
                    models.BigAutoField(
                        auto_created=True,
                        primary_key=True,
                        serialize=False,
                        verbose_name="ID",
                    ),
                ),
                (
                    "error_code",
                    models.IntegerField(default=400, verbose_name="Код ошибки"),
                ),
                (
                    "description_for_devs",
                    models.TextField(
                        verbose_name="Техническое описание ошибки для разработчиков"
                    ),
                ),
                (
                    "description_for_users",
                    models.TextField(
                        default="",
                        help_text="Можно показывать пользователю в диалоговом окне для ошибок",
                        verbose_name="Описание для UI",
                    ),
                ),
                (
                    "stacktrace",
                    models.TextField(blank=True, default="", verbose_name="Stacktrace"),
                ),
                (
                    "created_date",
                    models.DateTimeField(
                        default=django.utils.timezone.now,
                        editable=False,
                        verbose_name="Время ошибки",
                    ),
                ),
                (
                    "handled_gracefully",
                    models.BooleanField(
                        default=True,
                        verbose_name="Была ли ошибка обработана кодом (или поймана exception handler middleware'ом)",
                    ),
                ),
            ],
            options={
                "verbose_name": "Ошибка",
                "verbose_name_plural": "Ошибки",
            },
        ),
    ]
